File: unix/bin/search-path.pl
#!/usr/bin/perl

use File::Spec::Functions 'path', 'rel2abs'; # Makes script cross-platform

foreach my $pattern (@ARGV) {
    foreach my $dir (path()) {
        if (-d $dir) {
            chdir($dir);
            foreach my $match (glob("*$pattern*")) {
                print rel2abs($match), "\n";
            }
        }
    }
}

File: unix/bin/search-path.py
#!/usr/bin/env python

import glob
import os
import sys

for pattern in sys.argv[1:]:
    path = os.environ["PATH"]
    for dir in path.split(os.pathsep):
        if os.path.exists(dir):
            os.chdir(dir)
            for match in glob.glob("*" + pattern + "*"):
                print(os.path.join(dir, match))
